test/modules/Rfq.t.sol

```diff
@@ -0,0 +1,309 @@
+// SPDX-License-Identifier: GPL-3.0-only
+pragma solidity ^0.8.18;
+
+import {MatchingBase} from "test/shared/MatchingBase.t.sol";
+import {IActionVerifier} from "src/interfaces/IActionVerifier.sol";
+import {TradeModule, ITradeModule} from "src/modules/TradeModule.sol";
+import {IPerpAsset} from "v2-core/src/interfaces/IPerpAsset.sol";
+import {IBaseManager} from "v2-core/src/interfaces/IBaseManager.sol";
+import {MockDataReceiver} from "../mock/MockDataReceiver.sol";
+import {IManager} from "v2-core/src/interfaces/IManager.sol";
+import {ISubAccounts} from "v2-core/src/interfaces/ISubAccounts.sol";
+import {IDutchAuction} from "v2-core/src/interfaces/IDutchAuction.sol";
+import {OptionEncoding} from "lyra-utils/encoding/OptionEncoding.sol";
+
+import "forge-std/console2.sol";
+import "../../src/interfaces/IRfqModule.sol";
+
+contract RfqModuleTest is MatchingBase {
+  // Test Rfq
+  // - Can fill an order successfully
+  // - Can fill multiple orders successfully
+  // - perps are filled correctly
+  // - Duplicate assets can be filled
+  // - Cash can be filled, allowing transfers between accounts with fees
+  // - charges fees as expected
+  // - can submit feed data
+  // - Reverts in different cases
+  //  - mismatch of subaccounts provided and in action
+  //  - hashed order mismatch
+
+  function testFillSpotRfqOrder() public {
+    weth.mint(address(this), 10e18);
+    weth.approve(address(baseAsset), 10e18);
+    baseAsset.deposit(dougAcc, 10e18);
+
+    IRfqModule.TradeData[] memory trades = new IRfqModule.TradeData[](1);
+    trades[0] = IRfqModule.TradeData({asset: address(baseAsset), subId: 0, markPrice: 1500e18, amount: 10e18});
+    _submitRfqTrade(trades);
+
+    assertEq(subAccounts.getBalance(camAcc, baseAsset, 0), 10e18);
+    assertEq(subAccounts.getBalance(dougAcc, baseAsset, 0), 0);
+
+    assertEq(subAccounts.getBalance(camAcc, cash, 0), int(cashDeposit) - 15000e18);
+  }
+
+  function testFillOptionBox() public {
+    IRfqModule.TradeData[] memory trades = new IRfqModule.TradeData[](4);
+    trades[0] = IRfqModule.TradeData({
+      asset: address(option),
+      subId: OptionEncoding.toSubId(block.timestamp + 1 weeks, 1500e18, true),
+      markPrice: 300e18,
+      amount: 2e18
+    });
+    trades[1] = IRfqModule.TradeData({
+      asset: address(option),
+      subId: OptionEncoding.toSubId(block.timestamp + 1 weeks, 1500e18, false),
+      markPrice: -300e18,
```

Comment:
unsure why this test uses a negative option price?

Reply:
intentional, just to show paying for the received asset